# Proyecto Módulo 1
# Inversión en instrumentos financieros


## Simulación Matemática
(Grupo martes - viernes)

Por:

Rodrigo Martínez Gallardo Bertoldi if722268

Luis Guillermo Covarrubias Chi nt709142

1 de octubre de 2019

Profesor: Cristian Camilo Zapata Zuluaga


## Objetivos

### Objetivo General
### Objetivos específicos 

|Límite inferior|Límite superior|Cuota fija|% sobre excedente de límite inferior|
|-----|-----|-----|-----|
|0.01|6,942.20|0.00|1.92|
6,942.21|58,922.16|133.28|6.40|
|58,922.17|103,550.44|3,460.01|10.88|
|103,550.45 | 120,372.83 | 8,315.57 | 16.00|
|120,372.84|144,119.23|11,007.14|17.92|
|144,119.24|290,667.75|15,262.49|21.36|
|290,667.76|458,132.29|46,565.26|23.52|
|458,132.30|874,650.00|85,952.92|30.00|
|874,650.01|1,166,200.00|210,908.23|32.00|
|1,166,200.01|3,498,600.00|304,204.21|34.00|
|3,498,600.01|En adelante|1,097,220.21|35.00|

**Se tienen $1,000,000 para invertir en distintos instrumentos financieros.**

La cantidad a invertir de cada instrumento debe de ser menor o igual al 20% del total invertido para diversificar el riesgo.

La cantidad a invertir de los instrumentos que pagan arriba del 10% anual, debe de ser menor o igual al 50% del total invertido para evitar pagar mas del 10.88% de ISR (Impuesto Sobre la Renta).

La cantidad a invertir de los instrumentos a corto plazo tiene que ser mayor o igual a 30% y la cantidad a invertir de los instrumentos largo plazo tiene que ser igual al 25%

|numero|Tipo|Plazo|Rendimiento anual|
|-----|-----|-----|-----|
|1|Cete|corto|7.1%|
|2|Bono|mediano|6.65%|
|3|Bancario|mediano|7.43%|
|4|Fondo privado|largo|15.2%|
|5|Bono Argentino|corto|23.3%|

Con respecto a  los tipos de instrumentos financieros, obtenemos que cada uno es:

$x_1=Cete$

$x_2=Bono$

$x_3=Bancario$

$x_4=Fondo privado$

$x_5=Bono Argentino$


 **Se toma en cuenta los enunciados para poder definir las restricciones para la optmización, quedando de la siguiente manera:**

"La cantidad a invertir debe de ser menor o igual al 20% del total invertido para diversificar el riesgo"

- $x_1 \leq 0.20(1,000,000)$

"La cantidad a invertir de los instrumentos a corto plazo tiene que ser mayor o igual a 30%"

- $x_1+x_5 \geq 0.30(1,000,000)$

Pero para poder utilizarse, se tendrá que convertir el $\geq$ a $\leq$, por lo tanto...

- $-x_1+-x_5\leq -0.30(1,000,000)$

"La cantidad a invertir de los instrumentos que pagan arriba del 10% anual, debe de ser menor o igual al 50% del total invertido"

- $x_4+x_5 \leq 0.50(1,000,000)$



"La cantidad a invertir de los instrumentos largo plazo tiene que ser igual al 25%"

- $x_4=0.25(1,000,000)$

La función objetivo, definida por los datos dados,será:

$$0.071x_1+0.0665x_2+0.0743x_3+0.152x_4+0.233x_5$$

In [12]:
#Se importan las librerías que necesitaremos
import scipy.optimize as opt
import numpy as np

In [37]:
c = -np.array([.071,0.0665,0.0743,0.152,0.233])

A_ub = np.array([[1,0,0,0,0],[-1,0,0,0,-1],[0,0,0,1,1]])
b_ub = np.array([.2*1000000,-.3*1000000,.5*1000000])

A_eq = np.array([[0,0,0,1,1]])
b_eq = np.array([.25*1000000])

bounds = (0,200000)

In [38]:
solucion = opt.linprog(c,A_ub,b_ub,A_eq,b_eq,bounds)
solucion

     fun: -96560.0
 message: 'Optimization terminated successfully.'
     nit: 6
   slack: array([     0., 250000.,      0.,      0.,      0., 150000.,      0.,
            0.])
  status: 0
 success: True
       x: array([200000., 200000., 200000.,  50000., 200000.])

Por lo tanto, las cantidades que se deben invertir en cada instrumento, tomando en cuenta las restricciones, para obtener el mayor número de rendimiento posible, son:

- $x_1= \$200,000$
- $x_2= \$200,000$
- $x_3= \$200,000$
- $x_4= \$50,000$
- $x_5= \$200,000$

Y el mayor rendimiento anual que se obtendrá será de $ \$96,560 $